pkg/promutil/prometheus_test.go

```diff
@@ -71,6 +72,12 @@ func TestSanitize(t *testing.T) {
 }
 
 func TestPromStringTag(t *testing.T) {
+	originalValidationScheme := model.NameValidationScheme
+	model.NameValidationScheme = model.LegacyValidation
+	defer func() {
+		model.NameValidationScheme = originalValidationScheme
+	}()
+
```

Comment:
I think this should raise attention to the breaking changes that added utf8 support:
- https://github.com/prometheus/common/releases/tag/v0.62.0
- https://github.com/prometheus/client_golang/releases/tag/v1.21.0

These two dependencies are now merged on `master` (unreleased), meaning that yace can now potentially export labels with utf8 characters. If the remote write destination does not support utf8, this would result in breaking behaviour right? I'm wondering if we should stick to the old validation scheme for some more time and add a feature flag to enable the new scheme.

Reply:
I agree. Could we do this in a follow up?

Reply:
Can we do the part where we set the legacy validation in this PR? Once pandora's box is opened, I guess it's worth merging a fully usable solution.